src/lib.rs

```diff
@@ -162,7 +163,26 @@ pub trait WriteBase32 {
     }
 
     /// Write a single `u5`
-    fn write_u5(&mut self, data: u5) -> Result<(), Self::Err>;
+    fn write_u5(&mut self, data: u5) -> Result<(), Self::Err> { self.write(&[data]) }
+}
+
+/// Interface to write `u8`s into a sink
+///
+/// Like `std::io::Writer`, but because the associated type is no_std compatible.
+pub trait WriteBase256 {
+    /// Write error
+    type Err: fmt::Debug;
+
+    /// Write a `u8` slice
+    fn write(&mut self, data: &[u8]) -> Result<(), Self::Err> {
+        for b in data {
+            self.write_u8(*b)?;
+        }
+        Ok(())
+    }
+
+    /// Write a single `u8`
+    fn write_u8(&mut self, data: u8) -> Result<(), Self::Err> { self.write(&[data]) }
```

Comment:
I think it's fine, it's a pretty common idiom for a trait to have infinite recursion by default, so that you can implement either one of the methods. If you try to run it, it will crash immediately (actually, I think the compiler will even flag it).